format/FormatNexusEigerDLS16M.py

```diff
@@ -5,31 +5,29 @@
 import libtbx
 
 from dxtbx.format.FormatNexusEigerDLS import FormatNexusEigerDLS
+from dxtbx.format.nexus import h5str
 from dxtbx.masking import GoniometerMaskerFactory
 from dxtbx.model import MultiAxisGoniometer
 
 VALID_NAMES = {
-    b"DIAMOND BEAMLINE I03",
-    b"DIAMOND BEAMLINE I04",
+    # "long" names
+    "DIAMOND BEAMLINE I03",
+    "DIAMOND BEAMLINE I04",
+    # "short" names
+    "DLS I03",
+    "DLS I04",
 }
 
-LEGACY_NAMES = {b"I03", b"I04"}
+LEGACY_NAMES = {"I03", "I04"}
 
 
 class FormatNexusEigerDLS16M(FormatNexusEigerDLS):
     @staticmethod
     def understand(image_file):
         # Get the file handle
         with h5py.File(image_file, "r") as handle:
-            name = FormatNexusEigerDLS.get_instrument_name(handle)
-            if name is None:
-                return False
-            if name.upper() in LEGACY_NAMES:
-                return True
-            if name in VALID_NAMES:
-                return True
-
-        return False
+            name = h5str(FormatNexusEigerDLS16M.get_instrument_name(handle))
+            return name and name.upper() in VALID_NAMES | LEGACY_NAMES
```

Comment:
Is there a reason that "Legacy names" is split up here and not just part of "valid names" - and any real prospect of _not_ checking them further down the line? presumably there is data out there with old name we don't want to drop support for....